core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -268,6 +268,23 @@ private void replaceDataFiles(Iterable<DataFile> deletedDataFiles, Iterable<Data
     }
   }
 
+  protected int getParallelism(int parallelism, List<CombinedScanTask> combinedScanTasks) {
```

Comment:
Do we need to pass the whole `List<CombinedScanTask>` here ?  I think we only need the `numOfScanTasks` ?

Reply:
btw,  it seems to be a method which was designed for flink, Better to move it to flink implementation class ?

Reply:
+1 I think we can move options to Flink too.

Reply:
should we set the max parallelism for spark  RewriteDataFilesAction?

Reply:
Spark can handle large parallelism in default...

Reply:
I moved set max parallelism to flink RewriteDataFilesAction